apps/server/lib/lexical/server/task_queue.ex

```diff
@@ -119,21 +119,19 @@ defmodule Lexical.Server.TaskQueue do
     end
 
     defp write_error(id, message, code \\ :internal_error) do
-      error =
-        ResponseError.new(
-          code: code,
-          message: message
-        )
+      error = ResponseError.new(code: code, message: message)
 
       Transport.write(%{id: id, error: error})
     end
 
-    defp run_task(fun) when is_function(fun) do
-      Task.Supervisor.async_nolink(task_supervisor_name(), fun)
+    defp run_task(fun, mfa) when is_function(fun) do
+      task = Task.Supervisor.async_nolink(task_supervisor_name(), fun)
+      %Task{task | mfa: mfa}
```

Comment:
I'm not sure it matters, but the mfa in a task should match the [mfa type](https://hexdocs.pm/elixir/1.17.2/typespecs.html#built-in-types), which is `{module, fun, arity}` instead of `{module, fun, args}`.

So we could do something like:

```suggestion
    defp run_task(fun, {m, f, args}) when is_function(fun) do
      task = Task.Supervisor.async_nolink(task_supervisor_name(), fun)
      %Task{task | mfa: {m, f, length(args)}}
```